### PR TITLE
layers: Combine alias functions calls

### PR DESCRIPTION
**Pull request by spencer-lunarg**
This reduces a lot of duplication code with alias/promoted function hooks,

With the new `ErrorObject` we always can know what function the call came from and there is not reason to keep 2 paths now.

A future goal would be maybe just having the chassis calling the non-KHR/EXT function right away saving another function jump

### CODE REVIEW

File: layers/stateless/stateless_validation.h [hunk 1]
```diff
@@ -838,6 +826,9 @@ class StatelessValidation : public ValidationObject {
     bool manual_PreCallValidateCmdDispatchIndirect(VkCommandBuffer commandBuffer, VkBuffer buffer, VkDeviceSize offset,
                                                    const ErrorObject &errorObj) const;
 
+    bool manual_PreCallValidateCmdDispatchBase(VkCommandBuffer commandBuffer, uint32_t baseGroupX, uint32_t baseGroupY,
```
Comment (spencer-lunarg) [inline layers/stateless/stateless_validation.h:829]: example case where we were not validating things from the promoted `vkCmdDispatchBase` but were only for `vkCmdDispatchBaseKHR`

I think its not realistic to test every promotion, but instead to generate it in future as that is only one point of failure then

File: layers/sync/sync_validation.cpp [hunk 2]
```diff
@@ -6376,6 +6315,10 @@ bool SyncValidator::PreCallValidateCmdSetEvent2(VkCommandBuffer commandBuffer, V
     const auto *cb_context = &cb_state->access_context;
     if (!pDependencyInfo) return skip;
 
+    const auto *access_context = cb_context->GetCurrentAccessContext();
```
Comment (spencer-lunarg) [inline layers/sync/sync_validation.cpp:6318]: notice we had these 3 lines in `PreCallValidateCmdSetEvent2KHR` but not `PreCallValidateCmdSetEvent2`

File: layers/core_checks/cc_device.cpp [hunk 3]
```diff
@@ -634,10 +634,9 @@ bool CoreChecks::PreCallValidateGetPhysicalDeviceImageFormatProperties2KHR(VkPhy
                                                                            const VkPhysicalDeviceImageFormatInfo2 *pImageFormatInfo,
                                                                            VkImageFormatProperties2 *pImageFormatProperties,
                                                                            const ErrorObject &errorObj) const {
-    // Can't wrap AHB-specific validation in a device extension check here, but no harm
-    bool skip = ValidateGetPhysicalDeviceImageFormatProperties2ANDROID(pImageFormatInfo, pImageFormatProperties, errorObj);
-    skip |= ValidateGetPhysicalDeviceImageFormatProperties2(pImageFormatInfo, pImageFormatProperties, errorObj);
-    return skip;
+    return PreCallValidateGetPhysicalDeviceImageFormatProperties2(physicalDevice, pImageFormatInfo, pImageFormatProperties,
+                                                                  errorObj);
+    ;
```
Comment (artem-lunarg) [inline layers/core_checks/cc_device.cpp:639]: nit: ;

File: layers/sync/sync_validation.cpp (hunk 2 above)
Comment (artem-lunarg) [reply layers/sync/sync_validation.cpp:6318]: looks safe for me

**Review by artem-lunarg — APPROVED** [review]
LGTM!

**Pull request merged**